plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/_impl.py

```diff
@@ -200,18 +204,39 @@ def sweep(self, arguments: List[str]) -> None:
             returns = self.launcher.launch(overrides, initial_job_idx=self.job_idx)
             self.job_idx += len(returns)
             for trial, ret in zip(trials, returns):
-                study._tell(trial, optuna.trial.TrialState.COMPLETE, [ret.return_value])
+                if len(self.optuna_config.directions) == 1:
+                    study._tell(
+                        trial, optuna.trial.TrialState.COMPLETE, [ret.return_value]
+                    )
+                else:
+                    study._tell(
+                        trial, optuna.trial.TrialState.COMPLETE, ret.return_value
+                    )
```

Comment:
Will Optuna raise a sensible error if the dimension of the return value does not match the dimension of the directions?

Reply:
Thank you for pointing it out. The current `Study_tell` just report given values without checking them. I added some exception handling of return values including the dimension mismatch and type incompatibility in https://github.com/facebookresearch/hydra/pull/1294/commits/f1eb00018e05f0aaace8a51a85ffddcff4d7959e.